eng/build.sh

```diff
@@ -264,6 +269,14 @@ function BuildSolution {
     fi
   fi
 
+  if [[ "$test_runtime_async" == true ]]; then
+    export DOTNET_RuntimeAsync="1"
```

Comment:
Just to confirm, we don't need to clean up this env variable afterwards?

Reply:
I noticed that the other flags aren't cleared either, but I'm not sure why not. Seems wrong not to clear them. We do in `build.ps1`